sequencer/src/hotshot_commitment.rs

```diff
@@ -168,7 +197,12 @@ async fn sync_with_l1(
     // error. We will retry, and may end up changing the transaction we send if the contract state
     // has changed, which is one possible cause of the transaction failure. This can happen, for
     // example, if there are multiple commitment tasks racing.
-    contract_send(&txn).await?;
+    contract_send(&txn)
+        .await
+        .map_err(|e| SyncError::TransactionFailed {
+            err: e,
+            num_leaves: max_blocks,
```

Comment:
We should set `leaves.len()` here, rather than `max_blocks`. Otherwise we might think we have reached `max_blocks` and hit a gas limit error, when we haven't and have actually failed for some other transient error.

Note that as long as we keep failing, more leaves will accumulate until we hit `max_blocks`, so we will always _eventually_ start decreasing the soft limit, and make progress